libraries/rdiff_backup.rb

```diff
@@ -0,0 +1,135 @@
+                                   "-r #{new_resource.destination} "\
+                                   "-p #{new_resource.nrpe_period} "\
+                                   "-l #{new_resource.nrpe_transferred}"
+          end
+        end
+        secrets = ::Chef::EncryptedDataBagItem.load('rdiff-backup-secrets',
+                                                    'secrets')
+        user new_resource.owner
+        group new_resource.group unless new_resource.owner == new_resource.group
+        sudo new_resource.owner do
+          user new_resource.owner
+          group new_resource.group
+          commands ['/usr/bin/sudo rdiff-backup'\
+                    '--server --restrict-read-only /']
+        end
+        directory ::File.join('/home', new_resource.owner, '.ssh') do
+          owner new_resource.owner
+          group new_resource.group || new_resource.owner
+          mode 0700
+        end
+        ssh_user_private_key 'id_rsa' do
+          user new_resource.owner
+          key secrets['ssh-key']
+        end
+
+        directory '/var/log/rdiff-backup' do
+          owner new_resource.owner
+          group new_resource.group
+          mode 0755
+          recursive true
+        end
+
+        [new_resource.destination, ::File.join('/home',
+                                               new_resource.owner,
+                                               'exclude',
+                                               new_resource.fqdn),
+         ::File.join('/home',
+                     new_resource.owner,
+                     'scripts',
+                     new_resource.fqdn)
+        ].each do |d|
+          directory d do
+            owner new_resource.owner
+            group new_resource.group || new_resource.owner
+            recursive true
+          end
+        end
+        file ::File.join('/home',
+                         new_resource.owner,
+                         'exclude',
+                         new_resource.fqdn,
+                         new_resource.source.gsub('/', '_')) do
+          owner new_resource.owner
+          group new_resource.group || new_resource.owner
+          mode 0644
+          content new_resource.exclude.join("\n")
+        end
+        filename = ::File.join('/home',
+                               new_resource.owner,
+                               'scripts',
+                               new_resource.fqdn,
+                               new_resource.source.gsub('/', '_'))
+        template filename do
+          source 'job.sh.erb'
+          mode 0775
+          owner new_resource.owner
+          group new_resource.group
+          cookbook new_resource.cookbook
+          variables(
+            fqdn: new_resource.fqdn,
+            src: new_resource.source,
+            dest: new_resource.destination,
+            period: new_resource.retention_period,
+            suser: new_resource.owner,
+            cuser: new_resource.remote_user,
```

Comment:
What's the reason for the ``suser`` and ``cuser`` variable names? At first I thought it was a typo, but then I saw it here too. I wasn't sure what ``s`` or ``c`` refers to. Might be better to either document this or rename it so its less confusion down the road.